### PR TITLE
feat: [PL-32664]: add task id to sync task connection tests

### PR DESCRIPTION
**Pull request by xingchi-jin**
## Describe your changes
Expose task id in connection test results
This PR also refactored the connector test response part codes, removed redundant codes.
 - include connectionTestResult and task id into the connector test framework, as most of the connector tests has these two data types in response. 

Test
PR ENV:
image

Local
image

AWS connector is also successful
image

Dockerhub connector is successful
Screen Shot 2023-05-30 at 3 52 35 PM

JFrog connector successful
Screen Shot 2023-05-30 at 4 04 45 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46630)

### CODE REVIEW

**Comment by xingchi-jin** [comment]
trigger utAll

**Comment by xingchi-jin** [comment]
trigger checkstylecodeformat

**Comment by xingchi-jin** [comment]
trigger pmd

**Comment by xingchi-jin** [comment]
trigger utAll

**Comment by xingchi-jin** [comment]
trigger utAll

**Comment by xingchi-jin** [comment]
trigger utAll

**Comment by xingchi-jin** [comment]
trigger ut0

**Comment by xingchi-jin** [comment]
trigger ut1

**Comment by xingchi-jin** [comment]
trigger ut2

**Comment by xingchi-jin** [comment]
trigger ut3

**Comment by xingchi-jin** [comment]
trigger ut4

**Comment by xingchi-jin** [comment]
trigger ut0

**Comment by xingchi-jin** [comment]
trigger ut1

**Comment by xingchi-jin** [comment]
trigger utAll

**Comment by xingchi-jin** [comment]
trigger ut8

**Comment by xingchi-jin** [comment]
trigger ut7

**Comment by xingchi-jin** [comment]
trigger ut5

**Comment by xingchi-jin** [comment]
trigger ut5

**Comment by xingchi-jin** [comment]
trigger ut6

**Comment by krishi0408** [comment]
trigger uts

**Review by piyush-harness — DISMISSED** [review]
> Reviewed 2 of 23 files at r1, 3 of 29 files at r2, 1 of 1 files at r4, 46 of 49 files at r5, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46630)* status: 55 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by xingchi-jin** [comment]
trigger uts

File: 440-connector-nextgen/src/main/java/io/harness/connector/validator/scmValidators/AbstractGitConnectorValidator.java [hunk 1]
```diff
@@ -133,9 +133,9 @@ public ConnectorValidationResult validate(ConnectorConfigDTO connectorConfigDTO,
       return super.validateConnectorViaManager(
           connectorConfigDTO, accountIdentifier, orgIdentifier, projectIdentifier, identifier);
     } else {
-      GitCommandExecutionResponse gitCommandExecutionResponse = (GitCommandExecutionResponse) super.validateConnector(
-          connectorConfigDTO, accountIdentifier, orgIdentifier, projectIdentifier, identifier);
-      return buildConnectorValidationResult(gitCommandExecutionResponse);
+      var responseData =
+          super.validateConnector(connectorConfigDTO, accountIdentifier, orgIdentifier, projectIdentifier, identifier);
+      return responseData.getConnectorValidationResult();
```
Comment (devkimittal) [inline 440-connector-nextgen/src/main/java/io/harness/connector/validator/scmValidators/AbstractGitConnectorValidator.java:138]: why?

Comment (xingchi-jin) [reply 440-connector-nextgen/src/main/java/io/harness/connector/validator/scmValidators/AbstractGitConnectorValidator.java]: @devkimittal i have updated the codes for git connector. to continue use the buildConnectorValidationResult function. The new codes added is just to add a taskid into the connectionValidationResult

**Comment by xingchi-jin** [comment]
trigger uts

**Review by piyush-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r7, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46630)* status: 55 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by devkimittal — APPROVED** [review]

File: 440-connector-nextgen/src/main/java/io/harness/connector/validator/scmValidators/AbstractGitConnectorValidator.java (hunk 1 above)
Comment (devkimittal) [reply 440-connector-nextgen/src/main/java/io/harness/connector/validator/scmValidators/AbstractGitConnectorValidator.java]: Approved

**Comment by piyush-harness** [comment]
:lgtm:

**Review by jennyjames21 — APPROVED** [review]
> Reviewed 2 of 23 files at r1, 3 of 29 files at r2, 1 of 1 files at r4, 44 of 49 files at r5, 3 of 3 files at r6, 1 of 2 files at r7, 1 of 1 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46630)* status: Some of the checks are too old. Waiting on team(s): del-team.

**Comment by piyush-harness** [comment]
Force merging this , Too many approvers.

**Pull request merged**